### PR TITLE
Always disable requiring Foundation for '@objc' when building/typechecking interfaces

### PR DESCRIPTION
**Pull request by artemcm**
Today, this (`-disable-objc-attr-requires-foundation-module`) is an extra flag we add in `InterfaceSubContextDelegateImpl::inheritOptionsForBuildingInterface`, which does not seem right for all cases, as it is not inherited from the parent invocation. It should instead just be the default for these frontend actions.

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci smoke test

File: test/ModuleInterface/ExplicitInterfaceObjC.swiftinterface [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// swift-interface-format-version: 1.0
+// swift-module-flags: -module-name ExplicitInterfaceObjC -module-interface-preserve-types-as-written
```
Comment (tshortli) [inline test/ModuleInterface/ExplicitInterfaceObjC.swiftinterface:2]: Is `-module-interface-preserve-types-as-written` critical to the test case?

**Review by tshortli — APPROVED** [review]

File: test/ModuleInterface/ExplicitInterfaceObjC.swiftinterface (hunk 1 above)
Comment (artemcm) [reply test/ModuleInterface/ExplicitInterfaceObjC.swiftinterface:2]: Nope, will delete, thanks.

**Comment by artemcm** [comment]
@swift-ci smoke test and merge

**Pull request merged**